site/functions/server/main.go

```diff
@@ -0,0 +1,287 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/apex/gateway"
+	"github.com/google/go-github/v32/github"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+	"sigs.k8s.io/krew/pkg/constants"
+	krew "sigs.k8s.io/krew/pkg/index"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	orgName    = "kubernetes-sigs"
+	repoName   = "krew-index"
+	pluginsDir = "plugins"
+
+	urlFetchBatchSize = 40
+	cacheSeconds      = 60 * 60
+)
+
+var (
+	githubRepoPattern = regexp.MustCompile(`.*github\.com/([^/]+/[^/#]+)`)
+)
+
+type PluginCountResponse struct {
+	Data struct {
+		Count int `json:"count"`
+	} `json:"data"`
+	Error ErrorResponse `json:"error,omitempty"`
+}
+
+type pluginInfo struct {
+	Name             string `json:"name,omitempty"`
+	Homepage         string `json:"homepage,omitempty"`
+	ShortDescription string `json:"short_description,omitempty"`
+	GithubRepo       string `json:"github_repo,omitempty"`
+}
+
+type ErrorResponse struct {
+	Message string `json:"message,omitempty"`
+}
+
+type PluginsResponse struct {
+	Data struct {
+		Plugins []pluginInfo `json:"plugins,omitempty"`
+	} `json:"data,omitempty"`
+	Error ErrorResponse `json:"error"`
+}
+
+func githubClient(ctx context.Context) *github.Client {
+	var hc *http.Client
+
+	// if not configured, you should configure a GITHUB_ACCESS_TOKEN
+	// variable on Netlify dashboard for the site. You can create a
+	// permission-less "personal access token" on GitHub account settings.
+	if v := os.Getenv("GITHUB_ACCESS_TOKEN"); v != "" {
+		ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: v})
+		hc = oauth2.NewClient(ctx, ts)
+	}
+	return github.NewClient(hc)
+}
+
+func pluginCountHandler(w http.ResponseWriter, req *http.Request) {
+	_, dir, resp, err := githubClient(req.Context()).
+		Repositories.GetContents(req.Context(), orgName, repoName, pluginsDir, &github.RepositoryContentGetOptions{})
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		writeJSON(w, PluginCountResponse{Error: ErrorResponse{Message: fmt.Sprintf("error retrieving repo contents: %v", err)}})
+		return
+	}
+	yamls := filterYAMLs(dir)
+	count := len(yamls)
+	log.Printf("github response=%s count=%d rate: limit=%d remaining=%d",
+		resp.Status, count, resp.Rate.Limit, resp.Rate.Remaining)
+
+	var out PluginCountResponse
+	out.Data.Count = count
+
+	w.Header().Set("Cache-Control", fmt.Sprintf("public, max-age=%d", cacheSeconds))
+	writeJSON(w, out)
+}
+
+func writeJSON(w io.Writer, v interface{}) {
+	e := json.NewEncoder(w)
+	e.SetIndent("", "  ")
```

Comment:
Let's not do this and save a few bits on the wire.
If we need to look at this, it will most certainly be in a browser console and they print json objects always the same.
```suggestion
```